templates/nomad_custom_data.sh.tpl

```diff
@@ -15,7 +15,11 @@ NOMAD_DIR_BIN="${nomad_dir_bin}"
 CNI_DIR_BIN="${cni_dir_bin}"
 NOMAD_USER="nomad"
 NOMAD_GROUP="nomad"
-NOMAD_INSTALL_URL="${nomad_install_url}"
+# NOMAD_INSTALL_URL="${nomad_install_url}"
+PRODUCT="nomad"
+OS_ARCH="linux_$( dpkg --print-architecture )"
```

Comment:
Sorry to complicate things but the call to dpkg needs to be swapped for a function to handle different distros like amazon linux and other rhel varients.